features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/model/TimelineItem.kt

```diff
@@ -66,10 +66,9 @@ sealed interface TimelineItem {
 
     @Immutable
     data class GroupedEvents(
-        val expanded: Boolean,
         val events: ImmutableList<Event>,
     ) : TimelineItem {
-        // use first id with a suffix
-        val id = events.first().id + "_group"
+        // use last id with a suffix. Last will not change in cas of new event from backpagination.
+        val id = events.last().id + "_group"
```

Comment:
Using the last item avoid the expanded group to be collapsed if new event get added to the group during backward pagination - which is probably to most common pagination direction.